src/driver-asus.c

```diff
@@ -423,6 +436,31 @@ asus_driver_probe(struct ratbag_device *device)
 	dpi_count = ratbag_device_data_asus_get_dpi_count(device->data);
 	button_count = ratbag_device_data_asus_get_button_count(device->data);
 	led_count = ratbag_device_data_asus_get_led_count(device->data);
+	const int8_t *bm = ratbag_device_data_asus_get_button_mapping(device->data);
+
+	/* merge ButtonMapping configuration property with defaults */
+	for (unsigned int i = 0; i < button_count; i++) {
+		drv_data->button_mapping[i] = bm[i] != -1 ? bm[i] : ASUS_CONFIG_BUTTON_MAPPING[i];
+		drv_data->button_indices[i] = -1;
+	}
+
+	/* setup a lookup table for all defined buttons */
+	unsigned int button_index = 0;
+	for (unsigned int i = 0; i < ARRAY_LENGTH(ASUS_BUTTON_MAPPING); i++) {
+		struct asus_button *asus_button = &ASUS_BUTTON_MAPPING[i];
+
+		/* search for this button in the ButtonMapping by it's ASUS code */
+		for (unsigned int j = 0; j < button_count; j++) {
+			if (drv_data->button_mapping[j] == (int8_t) asus_button->asus_code) {
+				/* add button to indices array */
+				drv_data->button_indices[button_index] = j;
```

Comment:
With `-Wconversion` this gives an implicit conversion warning.
```suggestion
				drv_data->button_indices[button_index] = (int)j;
```